pkg/api/grafana_com_proxy.go

```diff
@@ -24,8 +29,101 @@ var grafanaComProxyTransport = &http.Transport{
 	TLSHandshakeTimeout: 10 * time.Second,
 }
 
-func ReverseProxyGnetReq(logger log.Logger, proxyPath string, version string) *httputil.ReverseProxy {
-	url, _ := url.Parse(setting.GrafanaComUrl)
+func (hs *HTTPServer) ListGnetPlugins(c *models.ReqContext) {
+	proxy := hs.ReverseProxyGnetReq(c.Logger, "/plugins")
+	proxy.Transport = grafanaComProxyTransport
+	director := proxy.Director
+	proxy.Director = func(r *http.Request) {
+		// Call director set with ReverseProxyGnetReq
+		director(r)
+		// Request no encoding to simplify modifying the response
+		r.Header.Set("Accept-Encoding", "")
+	}
+	writeResponse := func(r *http.Response, b []byte) {
+		r.Body = io.NopCloser(bytes.NewReader(b))
+		r.ContentLength = int64(len(b))
+		r.Header.Set("Content-Length", strconv.Itoa(len(b)))
+	}
+	proxy.ModifyResponse = func(r *http.Response) error {
+		if hs.AccessControl.IsDisabled() {
+			return nil
+		}
+
+		defer func() { _ = r.Body.Close() }()
```

Comment:
Nit. Should log if failing to close the body. Example:
https://github.com/grafana/grafana/blob/9f77bd4728ba86a897ff837f1ae94bf19517004e/pkg/api/dashboard_snapshot.go#L70-L74